### ISSUE TITLE
syscall: TestCloneNEWUSERAndRemapNoRootDisableSetgroups fails on CentOS 7

### ISSUE THREAD

**Issue by EduRam** [golang/go#26385]
On CentOS 7, had an error on test TestCloneNEWUSERAndRemapNoRootDisableSetgroups test.
On Fedora28, all tests are sucessfull.

There is a previous Issue on the same matter #11261.
But in my case the message is different: "invalid argument, output:"

### What version of Go are you using (`go version`)?

go version go1.10.3 linux/amd64

### Does this issue reproduce with the latest release?

yes. i tried tip

### What operating system and processor architecture are you using (`go env`)?

```
$ uname -a
Linux geoserver.localdomain 3.10.0-862.6.3.el7.x86_64 #1 SMP Tue Jun 26 16:32:21 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

```
$ cat /etc/centos-release
CentOS Linux release 7.5.1804 (Core)
```

```
$ getenforce
Disabled
```

```
$go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/scatex/.cache/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/scatex/go"
GORACE=""
GOROOT="/home/scatex/wrk/go"
GOTMPDIR=""
GOTOOLDIR="/home/scatex/wrk/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build047489229=/tmp/go-build -gno-record-gcc-switches"
```

### What did you do?

1. Checkout go source code
2. Execute command go/src/all.bash

### What did you expect to see?

No errors on test phase.

### What did you see instead?

```
...
...
ok      sync/atomic     0.067s
--- FAIL: TestCloneNEWUSERAndRemapNoRootDisableSetgroups (0.00s)
        exec_linux_test.go:116: Cmd failed with err fork/exec /usr/bin/whoami: invalid argument, output:
--- FAIL: TestCloneNEWUSERAndRemapNoRootSetgroupsEnableSetgroups (0.00s)
        exec_linux_test.go:156: Unprivileged gid_map rewriting with GidMappingsEnableSetgroups must fail
--- FAIL: TestEmptyCredGroupsDisableSetgroups (0.00s)
        exec_linux_test.go:164: fork/exec /usr/bin/whoami: invalid argument
FAIL
FAIL    syscall 0.044s
ok      testing 2.007s
...
...
...
ok      cmd/vet/internal/cfg    0.026s
2018/07/14 06:40:54 Failed: exit status 1
```

**Comment by ianlancetaylor** [comment]
CC @LK4D4

**Comment by EduRam** [comment]
Output for command:
$ strace -f go test syscall -test.run=TestCloneNEWUSERAndRemapNoRootDisableSetgroups

[strace.txt](https://github.com/golang/go/files/2195995/strace.txt)

**Comment by EduRam** [comment]
The test passes now, after I changed some kernel settings:

# grubby --args="namespace.unpriv_enable=1 user_namespace.enable=1" --update-kernel="/boot/vmlinuz-3.10.0-862.6.3.el7.x86_64"

# echo "user.max_user_namespaces=15076" >> /etc/sysctl.conf

But it was completely uninformed!
Just following generic recipes from web.

**Comment by EduRam** [comment]
On some other Issue, someone asked to run this command.
In my case, the error message is "Invalid Argument".
(the same message on failed test)

```
$ unshare -Ur
unshare: unshare failed: Invalid argument
```

**Comment by EduRam** [comment]
I just realized there was previous Issues about this ... sorry for duplication:

[https://go-review.googlesource.com/c/go/+/49311](https://go-review.googlesource.com/c/go/+/49311)

**Comment by LK4D4** [comment]
It looks like now CentOS 7 has unprivileged namespaces disabled by default with kernel parameter.
The thing to check would be if it is CentOS 7 and there is `namespace.unpriv_enable=1` in `/proc/cmdline`, but it's not the first time we add band-aid for CentOS 7. Maybe it's easier just to add some comment to file like "Please find how to enable unprivileged namespaces for your distribution if you want these tests to work properly".

**Comment by ianlancetaylor** [comment]
Is there some way we can detect CentOS and just skip the test?  Perhaps look in `/etc/lsb-release`?

**Comment by ianlancetaylor** [comment]
> Maybe it's easier just to add some comment to file like "Please find how to enable unprivileged namespaces for your distribution if you want these tests to work properly".

In general the tests have to pass for users who don't understand what the code does.  So while that would be easier for us, it would just be mystifying for most people installing Go.  If the test can't reliably pass, we have to just skip it.

**Comment by LK4D4** [comment]
@ianlancetaylor Yeah, apparently all centos releases have `/etc/centos-release` file, maybe we can just check its presence

**Comment by EduRam** [comment]
> In general the tests have to pass for users who don't understand what the code does.
It's me!

Can I have a try on the fix ?
(check if /etc/centos-release presence)

**Comment by ianlancetaylor** [comment]
@EduRam Sure, send a fix, if you can do it very soon for the 1.11 release.  Thanks.
